### DISCUSSION TITLE
Fetch deduping/caching from within a node module

### DISCUSSION THREAD

**Discussion by psikoi** [vercel/next.js#54969] [App Router]
Does fetch deduping and caching still happen if that fetch call is made from within an **installed node module**?

Example:

```
// page.tsx
import * as myAPISDK from '@myapi/sdk'

export function generateMetadata() {
    const data = await myAPISDK.getUsers()
    return { title: data.length };
}

export default function Page() {
    const data = await myAPISDK.getUsers()
    return <div>{data}</div>
}
```

```
// @myapi/sdk/index.js

export function getUsers() {
    return fetch(...)
}
```

**Comment by psikoi** [comment]
Just ran a couple of tests, it doesn't seem like it does

**Reply by psikoi** [reply]
Possibly related issue: https://github.com/vercel/next.js/discussions/54969

**Reply by aduarterengifo** [reply]
Hi, I'm also running into this, can you share how you're testing this?

**Reply by psikoi** [reply]
I looked at my server logs, and I would see that 3 requests were being made per page load (page render, layout render and generateMetadata)

The fix was to create a bunch of functions wrapped in React's cache function. So instead of just using the example code I provided above, the final solution ended up being closer to

```
// @myapi/sdk/index.js

export function getUsers() {
    return fetch(...)
}
```

```
// page.tsx
import * as myAPISDK from '@myapi/sdk'

export const getUsers = cache(() => {
  return myAPISDK.getUsers();
});

export function generateMetadata() {
    const data = await getUsers()
    return { title: data.length };
}

export default function Page() {
    const data = await getUsers()
    return <div>{data}</div>
}
```

Had to write 24 of these "wrapped-cache" functions for all of my API endpoints, but it would be nice if this was done automatically somehow.

**Reply by aduarterengifo** [reply]
Thanks for sharing! 

Would exporting the wrapper functions directly from `@myapi/sdk` also do the trick?

**Reply by psikoi** [reply]
Not sure, I haven't tried that
